rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,115 @@
+
+As a data-wrangler, I want to be able to succinctly enter the metadata I receive from the data contributor and have a clear understanding of what I need to do to ensure correct processing.
+
+As a schema developer, I want to ensure we are capturing and requiring, the metadata from a contributor to ensuring the pipelines can run appropriately.
+
+As a member of the pipelines team, I want to ensure that the pipelines process the correct data sets and I can easily access the metadata required to trigger the pipelines.  I do not want to be restricted in designing analysis based on how data was submitted.
+
+As a data consumer, I want to be able to find all data files that need to be processed together so that I can run my data processing pipeline. I do not want to be restricted in designing analysis based on how data was submitted.
+
+As a data consumer, I want to be confident that the HCA DCP has correctly processed all data files that belong together so that I know the matrices I receive have been generated correctly.
+
+As a data consumer, I need to know that a data set is done so that I don't download and use it until it is version-complete.
+
+## Detailed Design
+A new concept of *data group* is added to the DCP data model to address these issues.  A data group is defined as a set of specific versions of metadata and data that is complete and consistent by specified criteria.  A *data group* is not created until all its contents are complete.  Events are generated when a *data group* is created, updated, or deleted.
+
+A data group has a symbolic scope type that specifies what is represented by the group, as well a the set of criteria by which it is complete.  Most of the above use cases will require a *project submission* scope that indicates submission to a project is complete.
+
+Data groups are implemented as a new bundle type that contains a JSON file listing the FQIDs (UUIDs with versions) of all bundles that part of the data groups and the existing DSS subscription mechanism for notifications.
+
+A new schema *data_group* will be created that contains the fields:
+- *scope* - Symbolic name of the scope, for example, *PROJECT_SUBMISSION*.
+- *bundle_fqids* - List of FQIDs of bundles that are in the group.
+
+
+Initially, only the *PROJECT_SUBMISSION* scope will be implemented. *PROJECT_SUBMISSION* will indicate that the entirety of single project submission is version-complete and will be generated by ingesting on after all data bundles are committed to the DSS.  This is indicated that data in the project submission should be processed by downstream components.  Ingest is responsible for creating *PROJECT_SUBMISSION* *data groups* after the relevant bundles have been created.  Figure 1 shows a diagram of a project with multiple *data groups*.
```

Comment:
In a conversation with Nick, the original proposal was to create a PROJECT_SUBMISSION notification (using your language) every time a submission happened for that project. The notification would be a data group (nee DAPS bundle) sent through the datastore. However, a particular submission could just be a part of a dataset, or even just something like all the donors when the user will add other metadata later.. In our original conversation we decided (maybe implictly by some of us) that it didn't matter - you'd just get whatever the project happened to be - it's [just some collection of data and metadata](https://docs.google.com/document/d/1qlgIROPK4qESy9-Lwvva1ZWJ3S5EdZ0HWJiGGvt2JRM/edit#). But it sounds like you need project to be a richer concept. What would that be?

Really, my feeling is that to make this work as you're describing, ingests needs new first class entities to model data groups directly rather than trying to fit these onto submissions or projects. Those concepts would need to fit into the submitter's mental model which is where it becomes a UX question (@gabsie).

Reply:
I see this is more or less your first unresolved question where data group may may to experiment data sets.

Reply:
Yes, I believe it might be very useful to map this to a scientific concept and represent it in the metadata.  Experiment seems to be the right concept, however, that needs to be carefully defined.

Reply:
Yes, I would say a new entity is critical for this work to proceed unless we come up with a suitable definition for 'project'.

Reply:
The current definition of DGs would allow this all to work independently of the experimental,
however, I feel like it will be a lot less understandable that way.  Another layer of confusion.